controller/classesController.js

```diff
@@ -0,0 +1,143 @@
+const classesService = require("../services/classesService");
+//console.log 삭제 필요
+const getClassesList = async(req,res) => {
+  try{
+    let { topCategoryName, subCategoryName, sortBy, search} = req.query;
+    console.log(req.query);
+    const result = await classesService.getClassesList(
+      topCategoryName,subCategoryName, sortBy, search
+    );
+    return res.status(200).json({ result });
+  } catch (err) {
+    console.log(err);
+    return res.status(500).json({ message: err.message });
+  };
+};
+
+const getUpcomingClasses = async(req,res) => {
+  try{
+    const result = await classesService.getUpcomingClasses();
+    return res.status(200).json({ message: result});
+
+  } catch (err) {
+    console.log(err);
+    return res.status(500).json({ message: err.message });
+  };
+
+};
+const getClassesDetail = async(req, res) => {
+  try {
+    const classId = req.params.classId;
+    const result = await classesService.getClassDetail(classId);
+    return res.status(200).json({ message: result });
+  } catch (err) {
+    console.error(err);
+    return res.status(err.statusCode || 500).json({ message: err.message });
+  }
+};
+
+const getMyClassesList =  async(req,res) => {
+  try {
+    const userId = req.users.id;
+    const result = await classesService.getMyClassesList(userId);
+    return res.status(200).json({ message: result });
+  } catch (err) { 
+    console.error(err);
+    return res.status(err.statusCode || 500).json({ message: err.message });
+  };
+};
+
+const getHostClassesList =  async(req,res) => {
+  try {
+    const hostId = req.hosts.id;
+    const result = await classesService.getHostClassesList(hostId);
+    return res.status(200).json({ message: result });
+  } catch (err) {
+    console.error(err);
+    return res.status(err.statusCode || 500).json({ message: err.message });
+  };
+};
+const createClass = async(req,res) => {
+  try{
+    const hostId = req.hosts.id;
+    const result = await classesService.createClass(hostId,req);
```

Comment:
```suggestion
    const result = await classesService.createClass(hostId, ___);
```

request에서 필요한 내용만 선정하여 따로 service layer로 내려주세요.
req 객체를 핸들링하는 과정은 controller에서만 수행해야 하고, service layer에서는 가공해서는 안됩니다.

This is "Seperation Of Concern"

Reply:
깨달음을 얻었습니다